front-end-tools/src/Main.tsx

```diff
@@ -588,13 +660,29 @@ export default function Main(props: ConnectionProps) {
                                                             ? moduleReferenceDeployed
                                                             : moduleReferenceCalculated
                                                     );
-                                                    setInitName(contracts[0]);
+
+                                                    setDisplayContracts(contracts);
+                                                    setContractName(contracts[0]);
                                                 }
                                             }}
                                         />
                                         <span>{' Use Module from Step 1'}</span>
                                     </label>
                                 </div>
+                                {useModuleFromStep1 && (
+                                    <>
+                                        <br />
+                                        <div>
+                                            This checkbox autofilled the `moduleReference`, the `contractNames`, and the
```

Comment:
It's unclear from the user interface what "contractNames" refers to. The UI element has the header "Smart contract name". Similarly "moduleReference" is "Module reference".